src/batched/KokkosBatched_UTV_TeamVector_Internal.hpp

```diff
@@ -41,23 +41,24 @@ namespace KokkosBatched {
       matrix_rank = -1;
       TeamVectorQR_WithColumnPivotingInternal
       	::invoke(member,
-      		 m, m,
+      		 m, n,
       		 A, as0, as1,
       		 t, ts0,
       		 p, ps0,
       		 work,
       		 matrix_rank);
-      
+
       TeamVectorQR_FormQ_Internal
       	::invoke(member,
-      		 m, matrix_rank,
+      		 m, matrix_rank, matrix_rank,
       		 A, as0, as1,
       		 t, ts0,
       		 U, us0, us1,
       		 work);
+      member.team_barrier();
```

Comment:
parts of A used in forming U could be destroyed in the rank deficient case if some threads got ahead here and began executing TeamVectorSetLowerTriangularInternal.